src/containers/my-cooperations/accept-cooperation-close/AcceptCooperationClosing.tsx

```diff
@@ -1,28 +1,81 @@
 import { ErrorOutlineRounded } from '@mui/icons-material'
-import { Typography } from '@mui/material'
+import { Box, Typography } from '@mui/material'
 import { useTranslation } from 'react-i18next'
 import CooperationActionBanner from '~/containers/my-cooperations/cooperation-action-banner/CooperationActionBanner'
 import Button from '~/design-system/components/button/Button'
 
 import { styles } from './AcceptCooperationClosing.styles'
+import { useState } from 'react'
+import InputField from '~/design-system/components/input-field/InputField'
+import { InputFieldVariantEnum } from '~/design-system/components/input-field/InputField.constants'
+import useForm from '~/hooks/use-form'
 
 interface AcceptCooperationClosureProps {
   user: string
   onAccept: () => void
+  onReasonSubmit: (reason: string) => void
 }
 
 const AcceptCooperationClosing: React.FC<AcceptCooperationClosureProps> = ({
   user,
-  onAccept
+  onAccept,
+  onReasonSubmit
 }) => {
   const { t } = useTranslation()
+  const [isInputShown, setIsInputShown] = useState<boolean>(false)
+  const [isReasonSubmitted, setIsReasonSubmitted] = useState<boolean>(false)
+
+  const handleDecline = () => {
+    setIsInputShown(true)
+  }
+
+  const {
+    data,
+    errors,
+    trigger,
+    handleInputChange,
+    handleNonInputValueChange,
+    handleSubmit
+  } = useForm({
+    initialValues: { declineReason: '' },
+    validations: {
+      declineReason: (value) =>
+        value ? '' : t('cooperationDetailsPage.inputError')
+    },
+    onSubmit: (formData) => {
+      if (formData) {
+        onReasonSubmit(formData.declineReason)
+      }
+    }
+  })
+
+  const handleReasonSubmit = () => {
+    const isValid = trigger('declineReason')
+    handleSubmit()
+    if (isValid && !errors.declineReason) {
+      setIsReasonSubmitted(true)
+      setIsInputShown(false)
+    }
+  }
+
+  const isSubmitMessageShown =
```

Comment:
It is not boolean, name it `submitMessage`.
Also you can put it directly inside jsx with conditional rendering (`condition && <Component/>`) with out ternary operator and null
```suggestion
  const isSubmitMessageShown =
```